sdk/include/xdk/ntbasedef.h

```diff
@@ -787,7 +787,11 @@ extern "C++" { \
 #endif /* _M_AMD64 */
 
 /* C_ASSERT Definition */
+#if defined(__clang__) && defined(__cplusplus)
+#define C_ASSERT(expr) typedef char __C_ASSERT__[(expr) ? 1 : -1]
+#else
 #define C_ASSERT(expr) extern char (*c_assert(void)) [(expr) ? 1 : -1]
+#endif
```

Comment:
whatever works I guess.

Reply:
This one is not ideal because we get an unused typedef warning, but that's better than the error.

Reply:
What's the error with the extern version?

Reply:
```
sdk\include\ndk/peb_teb.h(513,1) :  error: function declaration cannot have variably modified type
C_ASSERT(FIELD_OFFSET(STRUCT(TEB), Vdm) == 0xF18);
^
sdk\include\psdk\winnt.h(809,38) :  note: expanded from macro 'C_ASSERT'
#define C_ASSERT(expr) extern char (*c_assert(void)) [(expr) ? 1 : -1]
                                     ^
```

Reply:
Fascinating. Sounds like it's having trouble because FIELD_OFFSET isn't using `__builtin_offsetof`. Your fix seems fine.